src/BuiltInTools/dotnet-watch/Internal/PhysicalConsole.cs

```diff
@@ -20,6 +24,25 @@ private PhysicalConsole()
             };
         }
 
+        public CancellationToken ListenForForceReloadRequest()
+        {
+            _cancellationTokenSource ??= new CancellationTokenSource();
+
+            Task.Run(() =>
+            {
+                var key = Console.ReadKey(intercept: true);
```

Comment:
Not sure how Console events work, but does this play fine with the CancelKeyPress event?

Reply:
Yup. That seems to take precedence unless you configure https://docs.microsoft.com/en-us/dotnet/api/system.console.treatcontrolcasinput?view=net-5.0